src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -675,16 +673,30 @@ static void AddPropertyToTypeDeclaration(CodeTypeDeclaration typeDeclaration, Pr
                 ? new CodeTypeReference(member.PropertyType.Name)
                 : member.PropertyType.CreateCodeTypeReference(member);
 
+            var propertyName = member.Name;
             var property = new CodeMemberProperty
             {
-                Name = member.Name,
+                Name = propertyName,
                 Type = propertyType,
                 Attributes = propertyAttributes,
                 CustomAttributes = CreateCustomAttributes(member, attributeFilter),
                 HasGet = hasGet,
                 HasSet = hasSet
             };
 
+            // DefaultMemberAttribute on type gets propagated to IndexerNameAttribute
+            var defaultMemberAttribute = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
```

Comment:
Was thinking of pulling this somewhere else but then I figured many of these generator methods have custom logic and this one doesn't make the already crappy situation much worse

Reply:
`SingleOrDefault` makes a hard assumption that no one will ever declare and use a constructor with two parameters on an attribute with the name `System.Reflection.DefaultMemberAttribute`, which is probably true.

Reply:
(option)

```cs
var defaultMember = typeDeclarationInfo.CustomAttributes
    .SingleOrDefault(x => x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute"
    ?.ConstructorArguments.SingleOrDefault()
    ?.Value as string;
```

Reply:
I find this train wreck harder to read but I can live with your option as well

Reply:
Now I remember again, 

![image](https://user-images.githubusercontent.com/174258/68569262-fb21fe00-045d-11ea-8f10-66b7cd917088.png)

```
  ApiGenerator.cs(691, 17): [CS0023] Operator '?' cannot be applied to operand of type 'CustomAttributeArgument'
```

Reply:
```
            var defaultMemberAttributeValue = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
                x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute")
                ?.ConstructorArguments.Select(x => x.Value).SingleOrDefault() as string;
```

Reply:
or even better

```
            var defaultMemberAttributeValue = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
                    x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute")
                ?.ConstructorArguments.Select(x => x.Value).OfType<string>()
                .SingleOrDefault();
```